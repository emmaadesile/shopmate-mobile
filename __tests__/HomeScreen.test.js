import React from "react";
import LandingPage from "../screens/HomeScreen";
import renderer from "react-test-renderer";

const props = {
  navigation: jest.fn()
};

describe("Home screen", () => {
  it("Renders landing page", () => {
    const component = renderer.create(<LandingPage {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
