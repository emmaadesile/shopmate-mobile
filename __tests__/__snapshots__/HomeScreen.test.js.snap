// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen Renders landing page 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    bottomButton={true}
    bounces={false}
    data={
      Array [
        Object {
          "backgroundColor": "#22C0EB",
          "image": 1,
          "imageStyle": Object {
            "height": 250,
            "width": 250,
          },
          "key": "screen1",
          "text": "You can shop for any product. Just search, add to your cart, and checkout anytime.",
          "textStyle": Object {
            "color": "#FFFFFF",
            "fontSize": 17,
          },
        },
        Object {
          "backgroundColor": "#20d2bb",
          "image": 1,
          "imageStyle": Object {
            "height": 250,
            "width": 250,
          },
          "key": "screen2",
          "text": "Shopping for your items has never been more convinient. ",
          "textStyle": Object {
            "color": "#FFFFFF",
            "fontSize": 17,
          },
        },
        Object {
          "backgroundColor": "#22bcb5",
          "image": 1,
          "imageStyle": Object {
            "height": 250,
            "width": 250,
          },
          "key": "screen3",
          "text": "We deliver aywhere. Enjoy delivery to doorstep in the shortest time possible.",
          "textStyle": Object {
            "color": "#FFFFFF",
            "fontSize": 17,
          },
        },
      ]
    }
    disableVirtualization={false}
    extraData={750}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onDone={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    paginationStyle={null}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showDoneButton={true}
    showNextButton={true}
    showsHorizontalScrollIndicator={false}
    slides={
      Array [
        Object {
          "backgroundColor": "#22C0EB",
          "image": 1,
          "imageStyle": Object {
            "height": 250,
            "width": 250,
          },
          "key": "screen1",
          "text": "You can shop for any product. Just search, add to your cart, and checkout anytime.",
          "textStyle": Object {
            "color": "#FFFFFF",
            "fontSize": 17,
          },
        },
        Object {
          "backgroundColor": "#20d2bb",
          "image": 1,
          "imageStyle": Object {
            "height": 250,
            "width": 250,
          },
          "key": "screen2",
          "text": "Shopping for your items has never been more convinient. ",
          "textStyle": Object {
            "color": "#FFFFFF",
            "fontSize": 17,
          },
        },
        Object {
          "backgroundColor": "#22bcb5",
          "image": 1,
          "imageStyle": Object {
            "height": 250,
            "width": 250,
          },
          "key": "screen3",
          "text": "We deliver aywhere. Enjoy delivery to doorstep in the shortest time possible.",
          "textStyle": Object {
            "color": "#FFFFFF",
            "fontSize": 17,
          },
        },
      ]
    }
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "height": 1334,
              "width": 750,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "space-around",
                },
                Object {
                  "backgroundColor": "#22C0EB",
                  "height": 1334,
                  "paddingBottom": 132,
                  "width": 750,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(255, 255, 255, .7)",
                    "fontSize": 26,
                    "fontWeight": "300",
                    "paddingHorizontal": 16,
                  },
                  undefined,
                ]
              }
            />
            <Image
              source={1}
              style={
                Object {
                  "height": 250,
                  "width": 250,
                }
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(255, 255, 255, .7)",
                    "fontSize": 16,
                    "fontWeight": "300",
                    "paddingHorizontal": 16,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 17,
                  },
                ]
              }
            >
              You can shop for any product. Just search, add to your cart, and checkout anytime.
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "height": 1334,
              "width": 750,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "space-around",
                },
                Object {
                  "backgroundColor": "#20d2bb",
                  "height": 1334,
                  "paddingBottom": 132,
                  "width": 750,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(255, 255, 255, .7)",
                    "fontSize": 26,
                    "fontWeight": "300",
                    "paddingHorizontal": 16,
                  },
                  undefined,
                ]
              }
            />
            <Image
              source={1}
              style={
                Object {
                  "height": 250,
                  "width": 250,
                }
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(255, 255, 255, .7)",
                    "fontSize": 16,
                    "fontWeight": "300",
                    "paddingHorizontal": 16,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 17,
                  },
                ]
              }
            >
              Shopping for your items has never been more convinient. 
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "height": 1334,
              "width": 750,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "space-around",
                },
                Object {
                  "backgroundColor": "#22bcb5",
                  "height": 1334,
                  "paddingBottom": 132,
                  "width": 750,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(255, 255, 255, .7)",
                    "fontSize": 26,
                    "fontWeight": "300",
                    "paddingHorizontal": 16,
                  },
                  undefined,
                ]
              }
            />
            <Image
              source={1}
              style={
                Object {
                  "height": 250,
                  "width": 250,
                }
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(255, 255, 255, .7)",
                    "fontSize": 16,
                    "fontWeight": "300",
                    "paddingHorizontal": 16,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 17,
                  },
                ]
              }
            >
              We deliver aywhere. Enjoy delivery to doorstep in the shortest time possible.
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "bottom": 16,
          "left": 16,
          "position": "absolute",
          "right": 16,
        },
        null,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 16,
          "justifyContent": "center",
          "margin": 16,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, .9)",
            "borderRadius": 5,
            "height": 10,
            "marginHorizontal": 4,
            "opacity": 1,
            "width": 10,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, .2)",
            "borderRadius": 5,
            "height": 10,
            "marginHorizontal": 4,
            "opacity": 1,
            "width": 10,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, .2)",
            "borderRadius": 5,
            "height": 10,
            "marginHorizontal": 4,
            "opacity": 1,
            "width": 10,
          }
        }
      />
    </View>
    <View
      style={false}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(0, 0, 0, .3)",
                "flex": 1,
                "justifyContent": "center",
              },
              false,
              Object {
                "backgroundColor": "#3F3D56",
                "borderRadius": 10,
                "elevation": 1,
                "height": 50,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.15,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 18,
                  "padding": 12,
                },
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Continue
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
